### PR TITLE
Simplify join-graph

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

This simplifies the join-graph with the following steps: 

- Remove the unused root node from the graph
- Make edges also operate on scoped symbols instead of references and use renames as source
- Add more tests for more logical plans

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/planner/optimizer/rule/EliminateCrossJoinTest.java [hunk 1]
```diff
@@ -430,5 +434,71 @@ public void test_resolve_multiple_aliases_with_same_name_in_logical_join_plan()
...
+
+    @Test
+    public void test_graph_with_order() throws Exception {
+        var order = new Order(a, new OrderBy(List.of(x)));
+        Symbol firstJoinCondition = e.asSymbol("a.x = b.y");
+        var firstJoin = new JoinPlan(order, b, JoinType.INNER, firstJoinCondition);
+        Symbol secondJoinCondition = e.asSymbol("b.y = c.z");
+        var join = new JoinPlan(firstJoin, c, JoinType.INNER, secondJoinCondition);
+
+        assertThat(join).isEqualTo(
```
Comment (matriv) [inline server/src/test/java/io/crate/planner/optimizer/rule/EliminateCrossJoinTest.java:451]: Have you considered using `hasOperators()` instead?
minor: if you stick to isEqual() you can maybe beautify it by using `"""` blocks instead.

**Review by matriv — APPROVED** [review]
Thx! Left minor comment for the tests.

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Pull request merged**